changes to MPI partitioner broke build

Unfortunately https://github.com/dealii/dealii/pull/5259 broke master, on macOS clang 9.0.0:
```
In file included from /Users/davydden/libs-sources/deal.ii/davydden/source/base/partitioner.cc:17:
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/partitioner.templates.h:355:24: error: call to 'abs' is ambiguous
                       std::abs(locally_owned_array[j] - *read_position) <=
                       ^~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:339:11: note: expanded from macro 'Assert'
    if (!(cond))                                                              \
          ^~~~
/Users/davydden/libs-sources/deal.ii/builds/master/source/base/partitioner.inst:111:45: note: in instantiation of function template specialization 'dealii::Utilities::MPI::Partitioner::import_from_ghosted_array_finish<unsigned long long>' requested here
 template void Utilities::MPI::Partitioner::import_from_ghosted_array_finish< unsigned long long int >(const VectorOperation::values ,
                                            ^
/usr/include/stdlib.h:137:6: note: candidate function
int      abs(int) __pure2;
         ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/stdlib.h:115:44: note: candidate function
inline _LIBCPP_INLINE_VISIBILITY long      abs(     long __x) _NOEXCEPT {return  labs(__x);}
                                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/stdlib.h:117:44: note: candidate function
inline _LIBCPP_INLINE_VISIBILITY long long abs(long long __x) _NOEXCEPT {return llabs(__x);}
                                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/math.h:693:1: note: candidate function
abs(float __lcpp_x) _NOEXCEPT {return ::fabsf(__lcpp_x);}
^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/math.h:697:1: note: candidate function
abs(double __lcpp_x) _NOEXCEPT {return ::fabs(__lcpp_x);}
^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/math.h:701:1: note: candidate function
abs(long double __lcpp_x) _NOEXCEPT {return ::fabsl(__lcpp_x);}
^
```
same with GNU 7.2.0 and Clang 5.0.0 tester

--- a/include/deal.II/base/partitioner.templates.h
+++ b/include/deal.II/base/partitioner.templates.h
@@ -283,7 +283,7 @@ namespace Utilities
 
 
 
-    namespace
+    namespace internal
     {
       // In the import_from_ghosted_array_finish we need to invoke abs() also
       // on unsigned data types, which is ill-formed on newer C++
@@ -379,8 +379,8 @@ namespace Utilities
                 // the face, values on this face obtained from each side might
                 // be different due to additions being done in different order.
                 Assert(*read_position == Number() ||
-                       get_abs(locally_owned_array[j] - *read_position) <=
-                       get_abs(locally_owned_array[j] + *read_position) *
+                       internal::get_abs(locally_owned_array[j] - *read_position) <=
+                       internal::get_abs(locally_owned_array[j] + *read_position) *
                        100000. *
                        std::numeric_limits<typename numbers::NumberTraits<Number>::real_type>::epsilon(),
                        typename LinearAlgebra::distributed::Vector<Number>::

--- a/include/deal.II/base/partitioner.templates.h
+++ b/include/deal.II/base/partitioner.templates.h
@@ -20,6 +20,8 @@
 #include <deal.II/base/partitioner.h>
 #include <deal.II/lac/la_parallel_vector.h>
 
+#include <type_traits>
+
 
 DEAL_II_NAMESPACE_OPEN
 
@@ -281,6 +283,31 @@ namespace Utilities
 
 
 
+    namespace
+    {
+      // In the import_from_ghosted_array_finish we need to invoke abs() also
+      // on unsigned data types, which is ill-formed on newer C++
+      // standards. To avoid this, we use std::abs on default types but
+      // simply return the number on unsigned types
+      template <typename Number>
+      typename std::enable_if<!std::is_unsigned<Number>::value,
+               typename numbers::NumberTraits<Number>::real_type>::type
+               get_abs (const Number a)
+      {
+        return std::abs(a);
+      }
+
+      template <typename Number>
+      typename std::enable_if<std::is_unsigned<Number>::value, Number>::type
+      get_abs (const Number a)
+      {
+        return a;
+      }
+
+    }
+
+
+
     template <typename Number>
     void
     Partitioner::import_from_ghosted_array_finish(const VectorOperation::values  vector_operation,
@@ -352,8 +379,8 @@ namespace Utilities
                 // the face, values on this face obtained from each side might
                 // be different due to additions being done in different order.
                 Assert(*read_position == Number() ||
-                       std::abs(locally_owned_array[j] - *read_position) <=
-                       std::abs(locally_owned_array[j] + *read_position) *
+                       get_abs(locally_owned_array[j] - *read_position) <=
+                       get_abs(locally_owned_array[j] + *read_position) *
                        100000. *
                        std::numeric_limits<typename numbers::NumberTraits<Number>::real_type>::epsilon(),
                        typename LinearAlgebra::distributed::Vector<Number>::

